src/main/java/bio/terra/workspace/service/datareference/utils/DataReferenceValidationUtils.java

```diff
@@ -7,19 +7,29 @@
 import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.regex.Pattern;
 import org.springframework.stereotype.Component;
 
 @Component
 public class DataReferenceValidationUtils {
 
   private ObjectMapper objectMapper;
   private DataRepoService dataRepoService;
+  final Pattern nameValidationPattern = Pattern.compile("^[a-zA-Z0-9][_a-zA-Z0-9]*$");
 
   public DataReferenceValidationUtils(ObjectMapper objectMapper, DataRepoService dataRepoService) {
     this.objectMapper = objectMapper;
     this.dataRepoService = dataRepoService;
   }
 
+  public void validateReferenceName(String name) {
+    if ((name.length() < 1 || name.length() > 63)
+        || !nameValidationPattern.matcher(name).matches()) {
```

Comment:
probably worth adding a check for null here. Even if it's not possible for name to be null in current usage, this method could be used elsewhere in the future.